### ISSUE TITLE
Load on demands empty children cause exception

### ISSUE THREAD

**Issue by ivan-kolesov** [mbraak/jqTree#858]
version 1.8.7

I am using on-demand loading with dataUrl option as a function.
The server generates a response, and in a certain case the server may return an empty array of children.
This causes an exception (in the source code) when in
```
this.doLoadDataFromUrl(null, node, () => {
  this.openNodeInternal(node, slide, onFinished);
});

```
openNodeInternal is called.

<img width="746" alt="image" src="https://github.com/user-attachments/assets/5b5ea619-ed17-4267-961f-2d3a906e7655">

In this state node.children has an empty array.
Is it possible to avoid trying to open a node when there are no children?

**Comment by mbraak** [comment]
I'll have a look.

**Comment by mbraak** [comment]
I cannot reproduce the error. When does the error occur? When the tree is initialized?

**Comment by ivan-kolesov** [comment]
> I cannot reproduce the error. When does the error occur? When the tree is initialized?

When I open on demand a child.

**Comment by mbraak** [comment]
Ah, ok. I wasn't able to reproduce the error that way.

Is there something different in your code that might trigger the error? Maybe an option?

**Comment by ivan-kolesov** [comment]
> Is there something different in your code that might trigger the error? Maybe an option?

Will try to make a sandbox

**Comment by ivan-kolesov** [comment]
Made a sandbox: https://codepen.io/ivan-kolesov/pen/poMqmVb
when try open `Theropods` error occurs. It seems that attribute `children` with empty value is the reason. Without `children` all correct.

**Comment by mbraak** [comment]
Thanks. I will have a look.

**Comment by mbraak** [comment]
Work in progress: #859

**Comment by mbraak** [comment]
The pr is merged. I plan to release a new version this week.

**Comment by mbraak** [comment]
The fix is in version 1.8.8.
